### PR TITLE
Fix waiting for service in clustered environment

### PR DESCRIPTION
**Pull request by worekleszczy**
Switch subscribing to service from message adapter to child actor - all context message adapters internally use single FuctionRef and this made cluster receptionist once watch over this actor twice with two different messages types which yields to runtime exception.

### CODE REVIEW

File: extension/src/main/scala/io/scalac/mesmer/extension/AkkaStreamMonitoring.scala [hunk 1]
```diff
@@ -290,6 +290,8 @@ final class AkkaStreamMonitoring(
 
   private def waitForStart(): Behavior[Command] = {
     setTimeout()
+    log.debug("Waiting for start")
```
Comment (jczuchnowski) [inline extension/src/main/scala/io/scalac/mesmer/extension/AkkaStreamMonitoring.scala:293]: maybe we should be more descriptive about what we are waiting for in the logs

File: extension/src/main/scala/io/scalac/mesmer/extension/util/GenericBehaviors.scala [hunk 2]
```diff
@@ -25,37 +26,32 @@ object GenericBehaviors {
...
+
         import context._
 
+        setLoggerName(classTag[I].runtimeClass)
+
         def start(): Behavior[Any] = {
 
-          val adapter = context.messageAdapter[Listing](identity)
+          /*
+            We use child actors instead of message adapters to avoid issues with clustered receptionist
```
Comment (jczuchnowski) [inline extension/src/main/scala/io/scalac/mesmer/extension/util/GenericBehaviors.scala:37]: We need a better description of the clustered receptionist problem.

File: extension/src/main/scala/io/scalac/mesmer/extension/util/GenericBehaviors.scala [hunk 3]
```diff
@@ -8,6 +8,7 @@ import akka.actor.typed.receptionist.ServiceKey
 import akka.actor.typed.scaladsl.Behaviors
 
 import scala.reflect.ClassTag
+import scala.reflect.classTag
```
[the remark this answers is not in the document]
Comment (worekleszczy) [reply extension/src/main/scala/io/scalac/mesmer/extension/util/GenericBehaviors.scala:11]: I wonder why scalafix didn't convert this to import `scala.reflect.{ClassTag, classTag}` 🤔

File: extension/src/main/scala/io/scalac/mesmer/extension/util/GenericBehaviors.scala (hunk 2 above)
Comment (worekleszczy) [reply extension/src/main/scala/io/scalac/mesmer/extension/util/GenericBehaviors.scala:37]: I'll come up with something

Comment (worekleszczy) [reply extension/src/main/scala/io/scalac/mesmer/extension/util/GenericBehaviors.scala]: I left a comment on why messageAdapters can't be use, but this is not specific to this fix but to working with receptionist. I guess there is no point of explaining reason there in depth as this solution is still valid use of actors.

**Review by jczuchnowski — APPROVED** [review]

**Pull request merged**